incubator/hnc/internal/reconcilers/object.go

```diff
@@ -370,16 +382,27 @@ func (r *ObjectReconciler) syncPropagated(ctx context.Context, log logr.Logger,
 		!reflect.DeepEqual(object.Canonical(inst), object.Canonical(srcInst)) ||
 		inst.GetLabels()[api.LabelInheritedFrom] != srcInst.GetNamespace() {
 		metadata.SetLabel(inst, api.LabelInheritedFrom, srcInst.GetNamespace())
-		return actionWrite, srcInst
 	}
+	return actionWrite, srcInst
+}
 
-	// The object already exists and doesn't need to be updated. This will typically happen when HNC
-	// is restarted - all the propagated objects already exist on the apiserver. Record that it exists
-	// for our statistics.
-	r.recordPropagatedObject(log, inst.GetNamespace(), inst.GetName())
-
-	// Nothing more needs to be done.
-	return actionNop, nil
+func (r *ObjectReconciler) getSelector(ctx context.Context, log logr.Logger, inst *unstructured.Unstructured) labels.Selector {
+	annot := inst.GetAnnotations()
+	selectorStr, ok := annot[api.AnnotationSelector]
```

Comment:
Just confirming: Is it that `AnnotationTreeSelector` is not yet implemented in the PR?

Reply:
Yes, it's not part of this pr